### PR TITLE
Add GLM Out of Sample Predictions Notebook 

### PR DESCRIPTION
**Pull request by juanitorduz**
I would like to add a small example to the docs. See https://github.com/pymc-devs/pymc-examples/issues/33 and the original post https://juanitorduz.github.io/glm_pymc3/. I am open for suggestions and changes to make this notebook useful for the community.

### CODE REVIEW

**Comment by juanitorduz** [comment]
> View / edit / reply to [this conversation](https://app.reviewnb.com/pymc-devs/pymc-examples/pull/37/discussion/) on ReviewNB
> 
> _MarcoGorelli commented on 2021-02-23T14:55:33Z_
> ----------------------------------------------------------------
> 
> > This could be the case when one-hot-encoding categorical variables.
> 
> This isn't something I've come across, and I'm not sure I see how that would happen as the one-hot encoding using training set information only - do you have a reference?

This is related to (this post)[https://dzone.com/articles/pandasscikit-learn-get-dummies-testtrain-sets], where once the model is deployed it might see new categories. One could try to use `handle_unknown = 'ignore'` if this is expected (and wanted), but then in the cross-validation step if we use `get_dummies` we are already showing the model what to expect. Maybe I can just delete this line as it is not very relevant for the post itself :)

**Comment by juanitorduz** [comment]
> View / edit / reply to [this conversation](https://app.reviewnb.com/pymc-devs/pymc-examples/pull/37/discussion/) on ReviewNB
> 
> _MarcoGorelli commented on 2021-02-23T14:55:34Z_
> ----------------------------------------------------------------
> 
> are you sure that
> 
> fpr, tpr, thresholds = roc_curve(
>     y_true=y_test, y_score=y_test_pred, pos_label=1, drop_intermediate=False
> )
> is correct? Looking at the sklearn docs , it seems that the second argument (y_score) should be the raw predictions p_test_pred rather than the predicted labels y_test_pred . Indeed, if you look at the examples in the sklearn docs, they pass `clf.predict_proba(X)[:, 1]`

You are totally right 🙈 ! Thank you so much! I corrected the typos and this error on [this commit](https://github.com/pymc-devs/pymc-examples/pull/37/commits/11be4ae6c3484cbf9b700c744204087496fe8a9e).

**Comment by MarcoGorelli** [comment]
Cool, thanks @juanitorduz - looks good to me, I would just advise to add a prior predictive check (even though it's not the main focus of your notebook, it's probably something to encourage where possible)

**Comment by juanitorduz** [comment]
@MarcoGorelli I added a subsection on prior predictive checks. Let me know what you think about it 🤓 .

**Comment by juanitorduz** [comment]
Thanks @OriolAbril ! I fixed it :)

File: examples/table_of_contents_examples.js [hunk 1]
```diff
@@ -17,6 +17,7 @@ Gallery.contents = {
     "case_studies/hierarchical_partial_pooling": "(Generalized) Linear and Hierarchical Linear Models",
     "generalized_linear_models/GLM-model-selection": "(Generalized) Linear and Hierarchical Linear Models",
     "generalized_linear_models/GLM-negative-binomial-regression": "(Generalized) Linear and Hierarchical Linear Models",
+    "generalized_linear_models/GLM-out-of-sample-predictions": "Generalized) Linear and Hierarchical Linear Models",
```
Comment (twiecki) [inline examples/table_of_contents_examples.js:20]:
```suggestion
    "generalized_linear_models/GLM-out-of-sample-predictions": "(Generalized) Linear and Hierarchical Linear Models",
```

**Comment by juanitorduz** [comment]
Thank you @ricardoV94 I fixed your suggestions! (I always have issues naming `patsy` 😆 )!

**Comment by juanitorduz** [comment]
Thank you all for  your suggestions and catching my typos 🙈  (it is hard to catch them with a 1-year old learning to walk around you 😄 )!

**Comment by twiecki** [comment]
@MarcoGorelli Yeah, when I read that I thought "this is a great suggestion" and I love prior checks but now seeing it implemented I don't think it really benefits much here and I can't think of a way to improve it, so I would argue to remove it again. What do you think?

**Review by ricardoV94 — CHANGES_REQUESTED** [review]
I left a comment about the true data model specification. It's not a blocker, but I would at least mention it in the notebook.

**Review by ricardoV94 — APPROVED** [review]
LGTM

**Comment by twiecki** [comment]
Thanks @juanitorduz!
